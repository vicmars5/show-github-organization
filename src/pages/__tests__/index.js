import React from "react"
import renderer from "react-test-renderer"

import Index from "../index"

describe('Index page', () => {
  it('renders correctly', () => {
    const data = {
      github: {
        organization: {
          name: 'The Debt Collective',
          description: 'Offering debtors a shared platform for organization, advocacy, and direct action. ',
          avatarUrl: 'https://avatars0.githubusercontent.com/u/13947390?v=4',
          url: 'https://github.com/debtcollective',
          repositories: {
            totalCount: 52,
            edges: [
              {
                node: {
                  id: 'MDEwOlJlcG9zaXRvcnkxNjA3OTM2Ng==',
                  name: 'strikedebt-wp',
                  description: 'Strike Debt's Wordpress website',
                  url: 'https://github.com/debtcollective/strikedebt-wp',
                  mentionableUsers: {
                    totalCount: 9
                  },
                  stargazers: {
                    totalCount: 1
                  }
                }
              },
              {
                node: {
                  id: 'MDEwOlJlcG9zaXRvcnkyNTk5Mjg0OQ==',
                  name: 'debtcollective-web',
                  description: '(DEPRECATED) Website for debtcollective.org',
                  url: 'https://github.com/debtcollective/debtcollective-web',
                  mentionableUsers: {
                    totalCount: 11
                  },
                  stargazers: {
                    totalCount: 12
                  }
                }
              },
              {
                node: {
                  id: 'MDEwOlJlcG9zaXRvcnkzMjQzNDQ0Mg==',
                  name: 'defense-to-repayment-wizard',
                  description: 'Defense to Repayment wizard frontend',
                  url: 'https://github.com/debtcollective/defense-to-repayment-wizard',
                  mentionableUsers: {
                    totalCount: 12
                  },
                  stargazers: {
                    totalCount: 0
                  }
                }
              }
            ]
          }
        }
      }
    }
    const tree = renderer
      .create(<Index data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
