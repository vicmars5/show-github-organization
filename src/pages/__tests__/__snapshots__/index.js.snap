// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page renders correctly 1`] = `
<div>
  <header
    className="border-b border-gray-400 bg-white p-3"
  >
    <div
      className="container mx-auto py-4 flex"
    >
      <img
        alt="The Debt Collective"
        className="w-20 h-20 mr-2"
        src="https://avatars0.githubusercontent.com/u/13947390?v=4"
      />
      <div>
        <a
          className="text-color-link"
          href="https://github.com/debtcollective"
        >
          <h1
            className="font-bold text-xl"
          >
            The Debt Collective
          </h1>
        </a>
        <p>
          Offering debtors a shared platform for organization, advocacy, and direct action. 
        </p>
      </div>
    </div>
  </header>
  <main
    className="container mx-auto p-3"
  >
    <article
      className="bg-white my-4 rounded p-3 border border-gray-400"
    >
      <a
        className="text-color-link"
        href="https://github.com/debtcollective/strikedebt-wp"
      >
        <h2
          className="text-lg font-bold"
        >
          strikedebt-wp
        </h2>
      </a>
      <p
        className=""
      >
        Strike Debt's Wordpress website
      </p>
      <p
        className="text-base pt-3"
      >
        <span
          className="pr-2"
        >
          <svg
            aria-hidden="true"
            className="h-4 inline-block pb-1"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
            />
          </svg>
           
          1
        </span>
        <span
          className="pr-2"
        >
          <svg
            aria-hidden="true"
            className="h-4 inline-block pb-1"
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
            />
          </svg>
           
          9
        </span>
      </p>
    </article>
    <article
      className="bg-white my-4 rounded p-3 border border-gray-400"
    >
      <a
        className="text-color-link"
        href="https://github.com/debtcollective/debtcollective-web"
      >
        <h2
          className="text-lg font-bold"
        >
          debtcollective-web
        </h2>
      </a>
      <p
        className=""
      >
        (DEPRECATED) Website for debtcollective.org
      </p>
      <p
        className="text-base pt-3"
      >
        <span
          className="pr-2"
        >
          <svg
            aria-hidden="true"
            className="h-4 inline-block pb-1"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
            />
          </svg>
           
          12
        </span>
        <span
          className="pr-2"
        >
          <svg
            aria-hidden="true"
            className="h-4 inline-block pb-1"
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
            />
          </svg>
           
          11
        </span>
      </p>
    </article>
    <article
      className="bg-white my-4 rounded p-3 border border-gray-400"
    >
      <a
        className="text-color-link"
        href="https://github.com/debtcollective/defense-to-repayment-wizard"
      >
        <h2
          className="text-lg font-bold"
        >
          defense-to-repayment-wizard
        </h2>
      </a>
      <p
        className=""
      >
        Defense to Repayment wizard frontend
      </p>
      <p
        className="text-base pt-3"
      >
        <span
          className="pr-2"
        >
          <svg
            aria-hidden="true"
            className="h-4 inline-block pb-1"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
            />
          </svg>
           
          0
        </span>
        <span
          className="pr-2"
        >
          <svg
            aria-hidden="true"
            className="h-4 inline-block pb-1"
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
            />
          </svg>
           
          12
        </span>
      </p>
    </article>
  </main>
  <footer
    className="p-3"
  >
    Built by Victor Diaz
  </footer>
</div>
`;
